### DISCUSSION TITLE
Should wire:key be unique for a page?

### DISCUSSION THREAD

**Discussion by chillbram** [livewire/livewire#6773] [1. Help] [answered]
Could it cause any problem if `wire:key`s are duplicated across multiple foreach/forelse loops? The documentation doesn't mention this specifically and it can be read as either meaning the keys have to be unique in their loop or the keys have to be unique in their template. As all examples use regular model ids for the items, I presume it is not a problem, but it would be nice to know that for sure.

**Comment by joshhanley** [comment, marked as answer]
@chillbram in V2 yes - it needs to be unique per page (as that was how morphdom handled them), in V3 no - they only need to be unique within a parent element (as that's how Alpine morph handles them - similar to Vue or Alpine's keys).

Hope this helps!

**Comment by chillbram** [comment]
Great! Thanks for the swift response!

**Reply by joshhanley** [reply]
No worries! 🙂
